src/runtime/recipe/handle.ts

```diff
@@ -220,6 +241,8 @@ export class Handle implements Comparable<Handle> {
   get ttl() { return this._ttl; }
   set ttl(ttl: Ttl) { this._ttl = ttl; }
   get isSynthetic() { return this.fate === 'join'; } // Join handles are the first type of synthetic handles, other may come.
+  get associators() { return this._associations.filter(a => a.associate === this); }
+  get associated() { return this._associations.filter(a => a.associator === this); }
```

Comment:
both directions are hosted in _associations? This makes cloning unlikely to succeed.

Reply:
During cloning we do 2 things to ensure it works:
1) We make associator responsible for re-creating these connections (akin to particle being responsible for recreating handle connections)
2) We clone non-synthetic handles first, and synthetic handles second.
There is a test that these connections are recreated correctly after a clone.

It's `it('can normalize and clone a recipe with a synthetic join handle'`

Reply:
ah yes, I see that you're using this.associated rather than this._associations. Add a comment explaining cloning strategy above?

Reply:
There is a comment explaining the cloning strategy in the recipe.ts, where we first clone regular handles and then synthetic ones. I think that's a better place for it, don't you think?

Reply:
yup, that makes sense to me.